dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/TrackerBundle.java

```diff
@@ -124,45 +113,50 @@
     /**
      * Tracked entities to import.
      */
-    @Builder.Default
     private List<TrackedEntity> trackedEntities = new ArrayList<>();
 
     /**
      * Enrollments to import.
      */
-    @Builder.Default
     private List<Enrollment> enrollments = new ArrayList<>();
 
     /**
      * Events to import.
      */
-    @Builder.Default
     private List<Event> events = new ArrayList<>();
 
     /**
      * Relationships to import.
      */
-    @Builder.Default
     private List<Relationship> relationships = new ArrayList<>();
 
     /**
      * Rule effects for Enrollments.
      */
-    @Builder.Default
     private List<RuleEffects> ruleEffects = new ArrayList<>();
 
     /**
      * Rule effects for Enrollments.
      */
-    @Builder.Default
     private Map<String, List<RuleEffect>> enrollmentRuleEffects = new HashMap<>();
 
     /**
      * Rule effects for Events.
      */
-    @Builder.Default
     private Map<String, List<RuleEffect>> eventRuleEffects = new HashMap<>();
 
+    private final Map<TrackerType, Map<String, TrackerImportStrategy>> resolvedStrategyMap;
+
+    public TrackerBundle()
+    {
+        this.resolvedStrategyMap = new EnumMap<>( TrackerType.class );
+
+        resolvedStrategyMap.put( TrackerType.RELATIONSHIP, new HashMap<>() );
+        resolvedStrategyMap.put( TrackerType.EVENT, new HashMap<>() );
+        resolvedStrategyMap.put( TrackerType.ENROLLMENT, new HashMap<>() );
+        resolvedStrategyMap.put( TrackerType.TRACKED_ENTITY, new HashMap<>() );
+    }
```

Comment:
I see, related to my previous comment, with "new" this map gets initialized.
However, maybe we could still use static builder and initialize the map in-line at line:148 and hide the constructor (by making it private). This way we force whoever will use TrackerBundle to use the static builder which will initialize the map. Instead, like it is now, someone could be unaware of this constructor and use the static builder, hence having a TrackerBundle with un-initialized resolvedStrategyMap

Reply:
Absolutely right.
Changed it